### PR TITLE
initial ziti-sdk-c wrap

### PR DESCRIPTION
**Pull request by ekoby**

### CODE REVIEW

**Comment by smilindave26** [comment]
Copyright header comments?

File: ziti/__init__.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+_patch_methods = dict(
+    create_connection = zitisock.create_ziti_connection,
+    getaddrinfo = zitisock.ziti_getaddrinfo
+)
+
+
+class monkeypatch(object):
+    def __init__(self):
+        print('setting ziti overrides')
```
[the remark this answers is not in the document]
Comment (ekoby) [reply ziti/__init__.py]: cleaned up

File: ziti/zitilib.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import ctypes
+import os
+from typing import Tuple
+
+_mod_path = os.path.dirname(__file__)
+ziti = ctypes.CDLL(_mod_path + '/libziti.so')
```
Comment (dovholuknf) [inline ziti/zitilib.py:20]: mmmm. how's this work on windows? fine?

Comment (dovholuknf) [reply ziti/zitilib.py:20]: probably should try with MacOS too?

File: ziti/zitisock.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import socket
+from typing import Tuple
+from ziti import zitilib
+
+py_socket = socket.socket
+
+class ZitiSocket(py_socket):
```
Comment (dovholuknf) [inline ziti/zitisock.py:21]: accept/bind/listen coming? do we need to support getsocketopts? does gethostbyname need to be overridden? maybe not needed yet?

File: ziti/context.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+class ZitiContext:
+    def __init__(self, ctx_p):
+        self._ctx = ctx_p
+
+    def connect(self, addr):
+        fd = zitilib.ziti_socket(socket.SOCK_STREAM)
+        service = bytes(addr, encoding = "utf-8")
+        zitilib.ziti_connect(fd, self._ctx, service)
+        return socket.socket(socket.AF_UNIX, socket.SOCK_STREAM, 0, fd)
```
Comment (dovholuknf) [inline ziti/context.py:26]: no udp yet?

**Review by dovholuknf — APPROVED** [review]
looks good to me as is... some questions that aren't related to accepting it as is. just don't know the overall goal ...

File: ziti/zitilib.py (hunk 2 above)
Comment (ekoby) [reply ziti/zitilib.py:20]: no windows/mac support yet

**Pull request merged**